projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistry.scala

```diff
@@ -62,37 +63,25 @@ private[lagom] class ProjectionRegistry(system: ActorSystem) {
       shardNames,
       ClusterDistributionSettings(system).copy(clusterShardingSettings = clusterShardingSettings)
     )
-
   }
 
   implicit val exCtx: ExecutionContext = system.dispatcher
   implicit val timeout: Timeout        = Timeout(1.seconds)
 
-  def startWorker(coordinates: WorkerCoordinates): Unit =
-    projectionRegistryRef ! StateRequestCommand(coordinates, Started)
-
   def stopWorker(coordinates: WorkerCoordinates): Unit =
-    projectionRegistryRef ! StateRequestCommand(coordinates, Stopped)
+    projectionRegistryRef ! WorkerRequestCommand(coordinates, Stopped)
 
-  def stopAllWorkers(projectionName: String): Unit = bulk(projectionName, stopWorker)
+  def startWorker(coordinates: WorkerCoordinates): Unit =
+    projectionRegistryRef ! WorkerRequestCommand(coordinates, Started)
+
+  def stopAllWorkers(projectionName: String): Unit =
+    projectionRegistryRef ! ProjectionRequestCommand(projectionName, Stopped)
 
-  def startAllWorkers(projectionName: String): Unit = bulk(projectionName, startWorker)
+  def startAllWorkers(projectionName: String): Unit =
+    projectionRegistryRef ! ProjectionRequestCommand(projectionName, Started)
 
   private[lagom] def getState(): Future[State] = {
     (projectionRegistryRef ? GetState).mapTo[State]
   }
 
-  private def bulk(projectionName: String, op: WorkerCoordinates => Unit): Unit = {
-    val eventualProjection: Future[Projection] = getState()
-      .map { state =>
-        state.findProjection(projectionName) match {
-          case None             => throw new ProjectionNotFound(projectionName)
-          case Some(projection) => projection
-        }
-      }
-    eventualProjection.map { projection =>
-      projection.workers.map(_.tagName).foreach(tagName => op(WorkerCoordinates(projectionName, tagName)))
-    }
-  }
```

Comment:
Bulk operations no longer translate to a `get`+`iterate` and, instead, map into a specific command the actor may process at will.